src/Esprima/JsxParser.cs

```diff
@@ -558,17 +558,19 @@ private Token PeekJsxToken()
 
     private void ExpectJsx(string value)
     {
-        var token = this.NextJsxToken();
-        if (token.Type != TokenType.Punctuator || token.Value is string val && val != value)
+        var token = NextJsxToken();
+        if (token.Type != TokenType.Punctuator
+            || (token.Value is string s && s != value || token.Value is JsxToken.JsxHolder holder && holder.Value != value))
```

Comment:
Don't get this change. If token is not a punctuator, the condition is true. Otherwise we know that it's a punctuator, so why check for `JsxHolder` in `token.Value`?

Reply:
I think this was a leftover from in-between state when I was hunting places where wrapping and unwrapping should happen, removed extra